lib/components/src/Accordion/Accordion.tsx

```diff
@@ -0,0 +1,200 @@
+    },
+    [open, setOpen, onOpen, itemMap]
+  );
+
+  const onItemExpand = useCallback(
+    (id: string) => {
+      const newOpen = { ...open };
+      let oldOpen: StateChange | undefined;
+
+      if (allowMultipleOpen) {
+        newOpen[id] = true;
+      } else {
+        Object.keys(open).forEach((key) => {
+          if (open[key]) {
+            oldOpen = { id: key, index: itemMap.current[key].index };
+          }
+
+          newOpen[key] = id === key;
+        });
+      }
+
+      if (oldOpen) {
+        onClose(oldOpen);
+      }
+
+      setOpen({ ...open, ...newOpen });
+      onOpen({ id, index: itemMap.current[id].index });
+    },
+    [open, setOpen, itemMap]
+  );
+
+  useEffect(() => {
+    if (Object.keys(itemMap.current).length === Children.count(children)) {
+      setOpen({ ...open, ...openMap.current });
+    }
+  }, [itemMap, openMap]);
+
+  return (
+    <AccordionContext.Provider
+      value={{ addToMap, onClose: onItemClose, onOpen: onItemExpand, open }}
+    >
+      <Wrapper data-sb-accordion="" bordered={bordered} rounded={rounded} lined={lined} {...rest}>
+        {children}
+      </Wrapper>
+    </AccordionContext.Provider>
+  );
+};
+
+type WrapperProps = {
+  bordered: boolean;
+  rounded: boolean;
+  lined: boolean;
+};
+
+const Wrapper = styled.ul<WrapperProps>`
+  overflow: hidden;
+  list-style: none;
+  margin: 0;
+  padding: 0;
+
+  ${({ theme, bordered }) =>
+    bordered
+      ? css`
+          border: 1px solid ${theme.appBorderColor};
+        `
+      : null}
+
+  ${({ theme, lined }) =>
+    lined
+      ? css`
+          & > [data-sb-accordion-item] {
+            border-bottom: 1px solid ${theme.appBorderColor};
+
+            &:last-child {
+              border-bottom: 0 none;
+            }
+
+            &[aria-expanded='true'] {
+              & > [data-sb-accordion-header] {
+                border-bottom: 1px solid ${theme.appBorderColor};
+              }
+            }
+          }
+        `
+      : null}
+
+  ${({ theme, rounded }) =>
+    rounded
+      ? css`
+          border-radius: ${theme.appBorderRadius}px;
+        `
+      : null}
+
+  ${({ theme }) => css`
+    background-color: ${theme.background.content};
+
+    & > [data-sb-accordion-item] {
+      & > [data-sb-accordion-header] {
+        padding: 16px;
+        cursor: pointer;
+        border: 1px solid transparent;
+
+        &:hover {
+          background-color: ${theme.background.hoverable};
+        }
+      }
+
+      & > [data-sb-accordion-body] {
+        font-size: 14px;
+      }
+
+      &[aria-expanded='true'] {
+        & > [data-sb-accordion-body] > [data-sb-accordion-body-inner] {
+          padding: 16px 16px 16px 42px;
+          background-color: ${theme.background.app};
+        }
+      }
+
+      &[data-sb-state-prevent-expander='true'] {
+        & > [data-sb-accordion-header] {
+          cursor: default;
+        }
+      }
+    }
+  `}
+`;
```

Comment:
We use the object style notation:

```suggestion
const Wrapper = styled.ul<WrapperProps>(
  ({ theme }) => ({
    overflow: 'hidden',
    listStyle: 'none',
    margin: 0,
    padding: 0,
    backgroundColor: theme.background.content,
  }),
  ({ theme, bordered }) =>
    bordered
      ? {
          border: `1px solid ${theme.appBorderColor}`,
        }
      : {},
  ({ theme, rounded }) =>
    rounded
      ? {
          borderRadius: theme.appBorderRadius,
        }
      : {},
  ({ theme, lined }) =>
    lined
      ? {
          '& > [data-sb-accordion-item]': {
            borderBottom: `1px solid ${theme.appBorderColor}`,
            '&:last-child': {
              borderBottom: '0 none',
            },
            '&[aria-expanded="true"]': {
              '& > [data-sb-accordion-header]': {
                borderBottom: `1px solid ${theme.appBorderColor}`,
              },
            },
          },
        }
      : {},

  ({ theme }) => ({
    '& > [data-sb-accordion-item]': {
      '& > [data-sb-accordion-header]': {
        padding: 16,
        cursor: 'pointer',
        border: '1px solid transparent',
        '&:hover': {
          backgroundColor: theme.background.hoverable,
        },
      },
      '& > [data-sb-accordion-body]': {
        fontSize: 14,
      },
      '&[aria-expanded="true"]': {
        '& > [data-sb-accordion-body] > [data-sb-accordion-body-inner]': {
          padding: '16px 16px 16px 42px',
          backgroundColor: theme.background.app,
        },
      },
      '&[data-sb-state-prevent-expander="true"]': {
        '& > [data-sb-accordion-header]': {
          cursor: 'default',
        },
      },
    },
  })
);
```

I'm not a huge fan of the super specific selectors. can you explain why:
```
& [data-sb-accordion-body-inner]
```
Is not good enough, and it has to be:
```
& > [data-sb-accordion-item] > [data-sb-accordion-body] > [data-sb-accordion-body-inner]
```

Do we expect accordions within accordions?